doc/http-router.md

```diff
@@ -0,0 +1,450 @@
+```
+
+**Custom Prefix Examples:**
+```clojure
+;; Mount under /api/db
+(def handler (router/create-ring-handler
+               :config config
+               :prefix "/api/db"))
+
+;; Mount under /v1/database
+(def handler (router/create-ring-handler
+               :config config
+               :prefix "/v1/database"))
+
+;; No prefix (root level)
+(def handler (router/create-ring-handler
+               :config config))  ; :prefix defaults to ""
+```
+
+**Development Mode**: For local development only, you can bypass authentication:
+```clojure
+(def dev-config {:dev-mode true})  ; WARNING: Never use in production!
+
+(def dev-handler
+  (router/create-ring-handler :config dev-config))
+```
+
+### 2. Distributed Mode
+
+Run the full HTTP server for distributed deployments:
+
+```bash
+# Run the standalone server
+clojure -A:http-server -m datahike.http.server config.edn
+```
+
+## Complete Minimal Example
+
+Save this as `server.clj` and run with `clojure server.clj`:
+
+```clojure
+(ns server
+  (:require [datahike.api :as d]
+            [datahike.http.router :as router]
+            [ring.adapter.jetty :refer [run-jetty]]))
+
+(def config
+  {:token    "securerandompassword"  ; Required for production
+   :dev-mode false})                  ; Set to true for local development only
+
+(defn -main []
+  ;; Create a test database for demo
+  (d/create-database {:store {:backend :mem :id "demo"}})
+
+  ;; Create handler with authentication and mount under /datahike
+  (let [handler (router/create-ring-handler
+                 :config config
+                 :prefix "/datahike")]
+
+    (println "\n=== Datahike Embedded Server ===")
+    (println "URL: http://localhost:8080/datahike")
+    (println "Auth: token required (see config)")
+    (println "\nTest with:")
+    (println "  curl -X GET http://localhost:8080/datahike/database-exists? \\")
+    (println "       -H \"Authorization: token securerandompassword\" \\")
+    (println "       -d '[{\"store\":{\"backend\":\"mem\",\"id\":\"demo\"}}]'")
+
+    (run-jetty handler {:port 8080 :join? false})))
+
+;; Connect from REPL
+(comment
+  ;; Start server
+  (-main)
+
+  ;; Connect with authentication (include prefix in URL)
+  (def conn (d/connect {:store {:backend :datahike-server
+                                 :url "http://localhost:8080/datahike"
+                                 :token "securerandompassword"}}))
+
+  ;; Use normally
+  (d/transact conn [{:name "Alice"}])
+  (d/q '[:find ?name :where [_ :name ?name]] @conn))
+```
+
+## API Functions
+
+### `create-routes`
+
+Main function for getting routes in various formats:
+
+```clojure
+(create-routes & {:keys [format prefix include-writers? middleware]})
+```
+
+Options:
+- `:format` - Output format (`:raw`, `:reitit`, `:compojure`, `:ring`)
+- `:prefix` - URL prefix for all routes (e.g., `"/datahike"` or `"/api/db"`)
+- `:include-writers?` - Include internal writer routes (default `false`)
```

Comment:
Why would we factor the writers out besides GraalVM compilation? If this is about write protection then we should probably think about how to handle more fine grained auth in general.